.cascade-code/left-nav-refactor/_components/left_nav/left_nav_format

```diff
@@ -0,0 +1,125 @@
+        #set ( $siblingTitle = $_EscapeTool.xml($sibling.getChild("display-name").value) )
+        #set ( $siblingType = $sibling.getName() )
+        #set ( $hideSibling = $_XPathTool.selectNodes($sibling, "dynamic-metadata[1][value='Yes'] or system-page[name = 'index']/dynamic-metadata[1][value='Yes']") )
+        #set ( $children = $_XPathTool.selectNodes($sibling, "system-page[name != 'index'] | system-folder") )
+        #set ( $hiddenChildren = $_XPathTool.selectNodes($sibling, "system-page[name != 'index']/dynamic-metadata[1][value='Yes'] | system-folder[dynamic-metadata[1][value='Yes']]") )
+        
+        #if ( $children.size() > $hiddenChildren.size() )
+            #set ( $siblingHasChildren = true )
+        #else
+            #set ( $siblingHasChildren = false )
+        #end
+         
+        #if ( $siblingType == "system-folder" && $hideSibling == [false] )
+        
+            #if ( $current == true )
+                <li class="active">
+            #else
+                <li>
+            #end
+
+            <a href="${siblingPath}/index" class="s-f-idx">
+                $siblingTitle
+                #if ( $siblingHasChildren && $current == false )
+                    <span class="plus">+</span>
+                #end
+            </a>
+            #if ( $siblingHasChildren )
+                #displayChildren( $children )
+            #end
+            </li>
+            
+        #elseif ( $siblingType == "system-page" && $hideSibling == [false] )
+
+            #if ( $current == true )
+                <li class="active">
+            #else
+                <li>
+            #end
+            
+            <a href="${siblingPath}" title="${siblingTitle}" class="s-p">
+                $siblingTitle 
+            </a>
+            </li>
+
+        #end
+    #end
+    </ul>
+#end
+
+#set ( $parentDisplayName = $parent.getChild("display-name") )
+
+### if fails to find parent page, don't do this LI...gives error (eg parent folder unindexed or not right metadata):
+#if ( $parentDisplayName.value )
+    #set ( $parentTitle = $_EscapeTool.xml( $parentDisplayName.value ) )
+    <div class="leftTitle">
+        <ul>
+            <li class="title">
+            <a href="$parent.getChild("link").value" title="${parentTitle}">
+                $parentTitle
+            </a>
+            </li>
+        </ul>
+    </div>
+#end
+
+#if( $current )       
+    #set ( $siblings = $_XPathTool.selectNodes($currentPage, "//system-folder[system-page[@current and name = 'index']] | //system-page[@current and name != 'index'] | //system-folder[system-page[@current and name = 'index']]/following-sibling::*[name != 'index'] | //system-folder[system-page[@current and name = 'index']]/preceding-sibling::*[name != 'index'] | //system-page[@current and name != 'index']/following-sibling::*[name != 'index'] | //system-page[@current and name != 'index']/preceding-sibling::*[name != 'index']") )
```

Comment:
This line could use some explanation.

Reply:
A bit nit picky, but anyway to make this line more readable by using multiple lines or multiple variables?

Reply:
This would be something Mandy might be able to explain. It's just what's used right now to get all siblings for any page no matter what its structure looks like. It's kind of complex because it has to go up one folder to get the siblings of the parent folder and all its children.